ql/src/java/org/apache/hadoop/hive/ql/io/orc/VectorizedOrcAcidRowBatchReader.java

```diff
@@ -1940,39 +2091,38 @@ public boolean isEmpty() {
     }
     @Override
     public void findDeletedRecords(ColumnVector[] cols, int size, BitSet selectedBitSet) {
-      if (rowIds == null || compressedOwids == null) {
+      if (rowIds == null || writeIds == null || writeIds.isEmpty()) {
         return;
       }
       // Iterate through the batch and for each (owid, rowid) in the batch
       // check if it is deleted or not.
 
       long[] originalWriteIdVector =
-          cols[OrcRecordUpdater.ORIGINAL_WRITEID].isRepeating ? null
-              : ((LongColumnVector) cols[OrcRecordUpdater.ORIGINAL_WRITEID]).vector;
+              cols[OrcRecordUpdater.ORIGINAL_WRITEID].isRepeating ? null
```

Comment:
Lets avoid changing the tabs/spaces below

Reply:
reverted